### DISCUSSION TITLE
_route vs route breaks layout

### DISCUSSION THREAD

**Discussion by akomm** [kiliman/remix-flat-routes#127] [Q&A] [answered]
As soon as you use the _route semantic, you break out of flat-routes. Forcing to only be able to use _layout implicitly forces be to nest more and more folders that I would like to have flat. I thought hybrid method is possible here. Did I miss something?

**Comment by akomm** [comment]
Examaple:
_user/_route.tsx <- not layout anymore after adding `_`
_user.stuff/_route.tsx 

This forces me to nest all _user.stuff into _user in order to add _user/_layout.tsx. It also forces me to move all the _user/_route.tsx co-located assets into the _user folder.

Its bad also bad because typically you see folders first, files last.

**Reply by kiliman** [reply, marked as answer]
Hmm.. interesting. With `remix-flat-routes` and the flat-folders convention (colocation), these are all _synonmous_: any file named _index, route, layout, page_ or any file starting with `_`.

So `_user/_route` should be treated as a parent layout for `_user.stuff/_route`. If not, then that is a bug.

Remember, `_layout` only has a special meaning with _hybrid_ routes in flat-files convention. Typically, you'd have `_user.tsx` and `_user.stuff.tsx`. If you want to user _hybrid_ routes it would be `_user+/_layout.tsx` and `_user+/stuff.tsx`. Essentially, `_user+/_layout.tsx` is equivalent to `_user.tsx`

Thanks again for these examples. It would help greatly if you could include your folder structure: `tree app/routes` and route config: `npx remix routes`.

**Reply by akomm** [reply]
I think I have a clue now with `npx remix routes` (nice to know, thanks!)

I think its a mistake on my side, because I split my action & loader into server.ts file and I've prefixed server.ts _server.ts to be always at the top with the route, but that makes server.ts the parent route:

![grafik](https://github.com/kiliman/remix-flat-routes/assets/2409779/911977bc-1ef5-48f5-a728-abed5da29823)

And its when it struck me, that as per documentation you can name routes _whateveryouwant.ts

**Reply by kiliman** [reply]
I see... you can always use `_route.tsx` for your route module, and `_route.server.tsx` as the server stuff.

```ts
// routes/_user/_route.tsx
import { action, loader } from './_route.server';
export { action, loader };
```

**Reply by akomm** [reply]
oh, I've confused the _route.server.tsx in my first reply with the route ID semantic. Yes that is a good idea.
